src/cpp/src/image_generation/vae.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/image_generation/generation_config.hpp"
+
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+namespace genai {
+
+class DiagonalGaussianDistribution {
+public:
+    explicit DiagonalGaussianDistribution(ov::Tensor parameters) {
+        ov::Shape shape = parameters.get_shape();
+        OPENVINO_ASSERT(shape[0] == 1, "Batch size must be 1");
+        shape[1] /= 2;
+
+        m_mean = ov::Tensor(parameters.get_element_type(), shape, parameters.data());
```

Comment:
`m_mean` doesn't own `parameters`'s data. If `parameters` is destroyed after `DiagonalGaussianDistribution` construction, `sample()`would read garbage from `mean_data`. You are just lucky to call `sample()` immediately.

Reply:
good catch, fixed